packages/amplify-graphql-auth-transformer/src/__tests__/owner-auth.test.ts

```diff
@@ -560,6 +560,58 @@ describe('owner based @auth', () => {
     expect(out.schema).toMatchSnapshot();
   });
 
+  test('@auth in bidirectional relational schema using references', () => {
+    const inputSchema = `
+      type Primary @model @auth(rules: [{ allow: owner }]) {
+        id: String! @primaryKey
+        owner: String
+        relatedMany: [RelatedMany] @hasMany(references: ["primaryId"])
+        relatedOne: RelatedOne @hasOne(references: ["primaryId"])
+      }
+
+      type RelatedMany @model @auth(rules: [{ allow: owner }]) {
```

Comment:
If not already covered in other tests, it would be nice to have the following additional case either in this or a separate schema:
- using the connected field i.e `primaryId` as the `ownerField`

Reply:
That's my next case :)